meteor/client/ui/Shelf/DashboardPanel.tsx

```diff
@@ -568,79 +570,53 @@ export function getUnfinishedPieceInstancesReactive(playlist: RundownPlaylist) {
 	let prospectivePieces: PieceInstance[] = []
 	const now = getCurrentTime()
 	if (playlist.activationId && playlist.currentPartInstanceId) {
-		prospectivePieces = PieceInstances.find({
-			startedPlayback: {
-				$exists: true,
-			},
-			playlistActivationId: playlist.activationId,
-			$and: [
-				{
-					$or: [
-						{
-							stoppedPlayback: {
-								$eq: 0,
-							},
-						},
-						{
-							stoppedPlayback: {
-								$exists: false,
-							},
-						},
-					],
-				},
-				{
-					$or: [
-						{
-							adLibSourceId: {
-								$exists: true,
-							},
-						},
-						{
-							'piece.tags': {
-								$exists: true,
-							},
-						},
-					],
-				},
-				{
-					$or: [
-						{
-							userDuration: {
-								$exists: false,
-							},
-						},
-						{
-							'userDuration.end': {
-								$exists: false,
-							},
-						},
-					],
-				},
-			],
-		}).fetch()
+		const partInstance = PartInstances.findOne(playlist.currentPartInstanceId)
+
+		if (partInstance) {
+			const rundown = Rundowns.findOne(partInstance.rundownId)
+			if (rundown) {
+				const showStyle = rundown.getShowStyleBase()
```

Comment:
Is the performance good with this? Seems like a lot of data to be pulled in and a lot of reactivity, even though, perhaps it doesn't have to be this reactive? I think it would be good to narrow the reactivity on the `Rundowns` at least? `rundown` seems to be pulled in only to get the ShowStyleBase, which I think could be piped in some other way maybe?

Reply:
pulling in the rundown did feel a bit excessive when I started this, I forgot to look at other routes for getting the showstylebase. Ill have a look into if we can pipe in a showstyle id, or if not ill trim the rundown down to just the necessary ids

Reply:
yep, two method calls up we always have a showStyleBase available. That is now passed in as a parameter